Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -682,105 +656,84 @@ namespace AzFramework
-                                        msg->m_senderTargetId = targetId;
-
-                                        m_inboxMutex.lock();
-                                        m_inbox.push_back(msg);
-                                        m_inboxMutex.unlock();
-                                    }
-                                }
-                                result = member->ReceiveBinary(m_tmpInboundBuffer.data(), static_cast<unsigned int>(m_tmpInboundBuffer.capacity()));
-                            }
-                        }
-                    }
-
-                    // Send
-                    size_t maxMsgsToSend = m_outbox.size();
-                    for (size_t iSend = 0; iSend < maxMsgsToSend; ++iSend)
-                    {
-                        GridMate::GridMember* peer = m_networkImpl->m_session->GetMemberById(m_outbox.front().first);
-                        if (peer)
-                        {
-                            peer->SendBinary(m_outbox.front().second.data(), static_cast<unsigned int>(m_outbox.front().second.size()), GridMate::Carrier::SEND_RELIABLE, GridMate::Carrier::PRIORITY_LOW);
-                        }
-
-                        m_outboxMutex.lock();
-                        m_outbox.pop_front();
-                        m_outboxMutex.unlock();
-                    }
+        void connect_target([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
+    {
+        for (auto& networkInterface : AZ::Interface<AzNetworking::INetworking>::Get()->GetNetworkInterfaces())
+        {
+            if (networkInterface.first == AZ::Name("TargetManagement"))
+            {
+                const AZ::CVarFixedString serverAddr = target_serveraddr;
+                const uint16_t serverPort = target_port;
+
+                AzNetworking::ConnectionId connId = networkInterface.second->Connect(
+                    AzNetworking::IpAddress(serverAddr.c_str(), serverPort, AzNetworking::ProtocolType::Tcp));
+                if (connId != AzNetworking::InvalidConnectionId)
+                {
+                    AzFrameworkPackets::Neighbor initPacket;
+                    initPacket.SetPersistentId(AZ::Crc32(Platform::GetPersistentName()));
+                    initPacket.SetCapabilities(Neighborhood::NEIGHBOR_CAP_LUA_VM | Neighborhood::NEIGHBOR_CAP_LUA_DEBUGGER);
+                    initPacket.SetPersistentName(Platform::GetPersistentName());
+                    initPacket.SetDisplayName(Platform::GetPersistentName());
+                    initPacket.SetIsJoin(true);
+                    networkInterface.second->SendReliablePacket(connId, initPacket);
                 }
             }
+        }
+    }
+    AZ_CONSOLEFREEFUNC(connect_target, AZ::ConsoleFunctorFlags::DontReplicate, "Opens a target management connection to a host");
 
-            AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(50));
+    TargetJoinThread::TargetJoinThread(int updateRate)
```

Comment:
An alternative for this could be to use an AZ::ScheduledEvent and then you wouldn't need the extra thread. Caveats to that are tickbus needs to be running for this to work (scheduled event chain off tick-bus), and the connect_target() and Stop() calls should be non-blocking.